### PR TITLE
Mixed typestable

### LINKED ISSUE #686: Type instability in pdf(::MixtureModel, x)
As of current type inference does not work for MixtureModels:
``` julia
d=MixtureModel([Normal()])
@code_warntype pdf(d,rand(d))
```
resulting in
```
[...]
end::Any
```

### PR DESCRIPTION
**Pull request by matbesancon**
Fixes #686 and adds docs
Also, `Categorical` where compared in tests without re-implementing `==`, which made comparison depend on vector comparison (non-elementwise)

### CODE REVIEW

File: src/mixtures/mixturemodel.jl [hunk 1]
```diff
@@ -1,26 +1,32 @@
...
+
+  - component(d, k):  return the k-th component
 
+  - probs(d):       return a vector of prior probabilities over components.
+"""
 abstract type AbstractMixtureModel{VF<:VariateForm,VS<:ValueSupport,C<:Distribution} <: Distribution{VF, VS} end
 
-struct MixtureModel{VF<:VariateForm,VS<:ValueSupport,C<:Distribution} <: AbstractMixtureModel{VF,VS,C}
+"""
+	MixtureModel{VF<:VariateForm,VS<:ValueSupport,C<:Distribution,CT<:Real}
```
Comment (simonbyrne) [inline src/mixtures/mixturemodel.jl]: Use spaces instead of tabs

File: src/mixtures/mixturemodel.jl [hunk 2]
```diff
@@ -1,26 +1,32 @@
...
+	A mixture of distributions, parametrized on:
```
Comment (simonbyrne) [inline src/mixtures/mixturemodel.jl]: remove this indentation (it will make it look like a code block)

**Comment by matbesancon** [comment]
Can someone check this one last time?

**Comment by matbesancon** [comment]
this gets merged in ~ 2hours if no objection

**Pull request merged**